lib/map/security_token/service.rb

```diff
@@ -8,15 +8,14 @@ module SecurityToken
     class Service < Common::Client::Base
       configuration Configuration
 
-      def token(application:, icn:)
+      def token(application:, icn:, cache: true)
         Rails.logger.info("#{config.logging_prefix} token request", { application:, icn: })
-        response = perform(:post,
-                           config.token_path,
-                           token_params(application, icn),
-                           { 'Content-Type' => 'application/x-www-form-urlencoded' })
-        sts_token = parse_response(response, application, icn)
-        Rails.logger.info("#{config.logging_prefix} token success", { application:, icn: })
-        sts_token
+        token = Rails.cache.fetch("map_sts_token_#{application}_#{icn}", expires_in: 5.minutes, force: !cache) do
+          cache = false
```

Comment:
I would probably create a new variable here, instead of overwriting the param. It makes sense what you did here and it looks like it works but I think that's will generally lead to confusing code. (so like `cached_response = true` at the top and then `cached_response = false` set in this block or something like that)